agent/internal/agent.go

```diff
@@ -279,11 +279,21 @@ func (a *agent) tlsConfig() (*tls.Config, error) {
 		}
 	}
 
+	var certs []tls.Certificate
+	if a.Options.Security.TLS.ClientTLSEnabled() {
```

Comment:
No need to do this check, since `ReadClientCertificate` is already doing it. Should just call that and check if the return value is nil.